javascript/ql/src/semmle/javascript/CharacterEscapes.qll

```diff
@@ -0,0 +1,88 @@
+/**
+ * Provides classes for reasoning about character escapes in literals.
+ */
+
+import javascript
+
+module CharacterEscapes {
+  /**
+   * Provides sets of characters (as strings) with specific string/regexp characteristics.
+   */
+  private module Sets {
+    string ordinaryEscapeChars() { result = "fnrtvux0\\" }
+
+    string regexpCharRangeChars() { result = "bBcdDsSwW" }
```

Comment:
Perhaps also include `p` and `P` for Unicode property classes.

Reply:
I think I'd prefer the terminology "character class" over "character range" here and elsewhere.